core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -95,6 +95,19 @@ sealed trait ConfigCursor {
   def asMap: Either[ConfigReaderFailures, Map[String, ConfigCursor]] =
     asObjectCursor.right.map(_.map)
 
+  final def at(pathSegment: PathSegment): Either[ConfigReaderFailures, ConfigCursor] = {
```

Comment:
Even though the compiler always chooses the most specific method to call, having this overloaded method in the public API of `ConfigCursor` might be a bit confusing to read. What do you think of making this definition private (or just inlining it in the other `at` definition)?

Reply:
* I named it `at` because it is a more general version of `atKey` and `atIndex`. `*Path` suffix didn't seem to add anything of value. But if you think it makes it more readable, happy to change the name back.
* I would keep the two methods separate because each does one specific thing, and I think that makes them easier to understand. I can make the non-varargs one private as the varargs one subsumes it for the users.